api.js

```diff
@@ -22,20 +19,27 @@ function init(page, observer, prevSpeed) {
 		}
 
 		if (result.isDone) {
-			page.close();
+			browser.close();
 			observer.complete();
 		} else {
-			setTimeout(init, 100, page, observer, result.speed);
+			setTimeout(init, 100, page, browser, observer, result.speed);
 		}
 	})
 	.catch(err => observer.error(err));
 }
 
 module.exports = () => new Observable(observer => {
-	driver.create({path: phantomjs.path})
-		.then(phantom => phantom.createPage())
-		.then(page => page.open('http://fast.com').then(() => {
-			init(page, observer);
-		}))
+	let browserInstance, pageInstance;
```

Comment:
Maintaining node >= 4 support here via promises. The code is a bit cleaner with async / await, but felt the compatibility was more important.